src/scripts/main.js

```diff
@@ -1,11 +1,29 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function correctly returns a promise that resolves when the specified event occurs on the element. However, be cautious about potential memory leaks if the event listeners are not removed after they are no longer needed. Consider using `once: true` in the `addEventListener` options to automatically remove the listener after it is triggered.